Improve program overview page.

mlehtima opened the pull request:
Improvements to program overview page and also added the ability to show program details from all list views through context menu the same way as from the first page.

mlehtima committed:
Improve program overview page.

mlehtima committed:
Add context menu to list delegate for showing program details.

mlehtima commented:
This pull request is ready for review.

joonne commented on qml/js/yleApi.js:
```diff
@@ -73,6 +73,7 @@ var mapPrograms = function(programs) {
         return {
             id: program.id,
             title: program.title && program.title.fi,
```
at some point we need to make changes to this in order to support localization properly for the API content also. Just a side note :)

mlehtima replied:
Yes that should be done at some point.

joonne commented on qml/pages/ProgramDelegate.qml:
```diff
@@ -3,10 +3,21 @@ import Sailfish.Silica 1.0
-    width: ListView.view.width
-    height: Math.ceil(9*contentWidth/3/16)
+    contentWidth: ListView.view.width
+    contentHeight: Math.ceil(9*contentWidth/3/16)
```
I'd prefer parentheses & spaces for this kind of calculations for clarity:

`contentHeight: Math.ceil(((9 * contentWidth) / 3) / 16)`

not an issue any way, looks maybe a bit clearer.

Also, where does 3 & 16 come from, could they be constants?

joonne commented on qml/pages/ProgramDelegate.qml:
```diff
@@ -3,10 +3,21 @@ import Sailfish.Silica 1.0
+    menu: ContextMenu {
+        MenuItem {
+            text: qsTr("Show program info")
+            onClicked: {
```
block could be removed as it is not used?

joonne commented on qml/pages/ProgramOverviewPage.qml:
```diff
@@ -17,32 +17,70 @@ Page {
         anchors.fill: parent
         contentHeight: column.height
 
+        PullDownMenu {
```
this is cool :)

joonne commented on qml/pages/ProgramOverviewPage.qml:
```diff
@@ -17,32 +17,70 @@ Page {
+                color: Theme.primaryColor
+                text: program.itemTitle != program.title ? qsTr("Episode") + ": " + program.itemTitle : ""
+                truncationMode: TruncationMode.Fade
+                visible: text != ""
```
!==

joonne commented on qml/pages/ProgramOverviewPage.qml:
```diff
@@ -17,32 +17,70 @@ Page {
+                color: Theme.primaryColor
+                text: YleApi.formatProgramDetails(program.seasonNumber, program.episodeNumber)
+                truncationMode: TruncationMode.Fade
+                visible: text != ""
```
!==

mlehtima commented:
Changed code based on comments. Calculation of contentHeight in ProgramDelegate was indeed something really odd, I replaced it with a value from the theme.

joonne commented:
looks good, I'll merge. Apparently GitHub offers squash & merge from the UI also :)

mlehtima commented:
Actually the code was in two commits on purpose, those were changing different parts of the code.

joonne commented:
sorry, my bad! :/ I did not pay attention enough then. Although I think I noticed that but I just got carried away with the automatic squash feature...
